src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -298,6 +298,15 @@ public final String getLine(int index) {
         return context.get().fileContents.getLine(index);
     }
 
+    /**
+     * Returns name of the file.
+     *
+     * @return filename.
+     */
+    public final String getFileName() {
+        return context.get().fileContents.getFileName();
```

Comment:
as we introduce new API method, should we make special issue on this to make it more visible/searchable to users ?

as linked issue is too obscure.

Reply:
created https://github.com/checkstyle/checkstyle/issues/11507

Reply:
Please use this issue in commit

Reply:
done